src/utils/dom.js

```diff
@@ -55,17 +55,25 @@ export const style = (root, src) => new Promise((resolve, reject) => {
  * Creates a new Script DOM element
  * @param {Element} root Root node
  * @param {string} src Source
+ * @param {Object} [options={}] Options
  * @return {Promise<StyleElement>}
  */
-export const script = (root, src) => new Promise((resolve, reject) => {
+export const script = (root, src, options = {}) => new Promise((resolve, reject) => {
+  const opts = {
+    async: false,
+    defer: false,
+    onerror: (err) => reject(err),
+    onload: () => resolve(el),
+    ...options,
+  };
+
   const el = document.createElement('script');
   el.onreadystatechange = function() {
     if ((this.readyState === 'complete' || this.readyState === 'loaded')) {
       resolve(el);
     }
   };
-  el.onerror = (err) => reject(err);
-  el.onload = () => resolve(el);
+  Object.assign(el, opts);
   el.src = src;
```

Comment:
Also, you can stick the `src` into this object as well. If combined with my previous suggestion it would not be possible to provide it via options as it will be explicitly set from function argument.